### PR TITLE
Feat tag release

### PR DESCRIPTION
**Pull request by CIGbalance**
Github release example: https://github.com/numbbo/coco-experiment/releases/tag/v2.6.101a7
Pypi: https://pypi.org/project/coco-experiment/2.6.101a7/#files

### CODE REVIEW

File: .github/workflows/tag_release.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: tag-release
+
+on:
+  push:
+    # Pattern matched against refs/tags
+    tags:        
+      - '**' # This includes hierarchical tags
```
Comment (olafmersmann) [inline .github/workflows/tag_release.yml:7]: Should we filter von tags that start with a v? I.e. 'v*' maybe?

File: .github/workflows/tag_release.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  python-wheels:
+    needs: source-build
+    uses: ./.github/workflows/test_python.yml
+    with:
+      build_only: "true"
+
+  release-python:
+    needs: python-wheels
+    if: ( success() || failure() )
```
Comment (olafmersmann) [inline .github/workflows/tag_release.yml:25]: Why does this continue on failure?

Comment (CIGbalance) [reply .github/workflows/tag_release.yml]: Fixed - this was so I could continue despite macos wheels failing

File: .github/workflows/tag_release.yml (hunk 1 above)
Comment (CIGbalance) [reply .github/workflows/tag_release.yml]: done

**Review by olafmersmann — APPROVED** [review]
LGTM

**Pull request merged**